assertj-core/src/main/java/org/assertj/core/condition/NestableCondition.java

```diff
@@ -138,16 +138,16 @@ public static <ACTUAL, NESTED> Condition<ACTUAL> nestable(String descriptionPref
    * @param <ACTUAL> the type of object the resulting condition accepts
    */
   @SafeVarargs
-  public static <ACTUAL> Condition<ACTUAL> nestable(String descriptionPrefix, Condition<ACTUAL>... conditions) {
-    return new NestableCondition<>(descriptionPrefix, stream(conditions));
+  public static <ACTUAL> Condition<ACTUAL> nestable(String descriptionPrefix, Condition<? super ACTUAL>... conditions) {
+    return new NestableCondition<ACTUAL, ACTUAL>(descriptionPrefix, stream(conditions));
   }
 
-  private NestableCondition(String descriptionPrefix, Stream<Condition<NESTED>> conditions, Function<ACTUAL, NESTED> extractor) {
+  private NestableCondition(String descriptionPrefix, Stream<Condition<? super NESTED>> conditions, Function<ACTUAL, NESTED> extractor) {
```

Comment:
Function<a, n> should be Function<? … a, ? … n> as well

Reply:
Hi @vlsi good point, thanks :+1: .

I've changed it for the function's input, but I can't find an example where changing the boundary of the function output is required :thinking: .

Reply:
`Function<ACTUAL, NESTED>` means function that `consumes ACTUAL` and `produces NESTED`.
With Java, you have to put `extends` and `super` all over the place (== in parameter declaration and return types).

For instance, `Function<A, B>` must always (except small edge cases) should be used as `Function<?  super A, ? extends B>`.
Then the method would accept `Function<.., ...>` that does not exactly match `Function<ACTUAL, NESTED>`.

See https://gist.github.com/vlsi/3e7c6a0aa3f13ec0cfc94c0f7447ceb7

Reply:
Thanks :+1: